File: s5-debug/files/fib.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include "fibonacci.h"

// https://ca.wikipedia.org/wiki/Leonardo_de_Pisa

int main(int argc, char * argv[])
{
   char c;
   if (argc != 2) {
      fprintf(stderr, "Us: %s <number>\n"
  "   where <number> is the number from which to compute fibonacci(number).\n",
       argv[0]);
      exit(1);
   }

   unsigned long num = strtol(argv[1], NULL, 0);
   unsigned long result = fibonacci(num);
   printf ("fibonacci(%lu) = %lu\n", num, result);
   //read(0, &c, 1);

   return 0;
}

File: s5-debug/files/fib.py
#!/usr/bin/python3
import sys

# import the external fib function
import fibonacci       # reads the file fibonacci.py

argc = len(sys.argv)

if (argc != 2) :
  print("Us:", sys.argv[0], "<number>", file=sys.stderr)
  print(
   "   where <number> is the number from which to compute fibonacci(number).",
   file=sys.stderr)
#  quit(1)
else :
  num = int(sys.argv[1])
  result = fibonacci.fibonacci(num)     # call function fibonacci in fibonacci.py
  print("fibonacci(", num, ") =", result)
#  input('Press Return to finish')

